build-gluster-org/scripts/generic_package.sh

```diff
@@ -0,0 +1,200 @@
+	case ${series} in
+	  "3.12")
+    	    debuild_key=${debuild_keys[0]}
+    	    pbuild_key=${pbuild_keys[0]}
+    	    ;;
+  	  "4.0")
+            debuild_key=${debuild_keys[1]}
+            pbuild_key=${pbuild_keys[1]}
+            ;;
+          "4.1")
+            if [ "$flavor" == "stretch" ]; then
+                 debuild_key=${pbuild_keys[2]}
+            else
+                 debuild_key=${debuild_keys[2]}
+            fi
+            pbuild_key=${pbuild_keys[2]}
+            ;;
+          "5" | "6" | "7")
+            if [ "$flavor" == "stretch" ]; then
+                 debuild_key=${pbuild_keys[3]}
+            else
+                 debuild_key=${debuild_keys[3]}
+            fi
+            pbuild_key=${pbuild_keys[3]}
+        esac
+else
+	echo "Exiting: OS should be debian or ubuntu. Please provide the right one"
+	exit
+fi
+
+mkdir ${os}-${flavor}-Glusterfs-${version}
+
+cd ${os}-${flavor}-Glusterfs-${version}
+
+mkdir build packages
+
+echo "Building glusterfs-${version}-${release} for ${flavor}"
+
+cd build
+
+TGZS=(`ls ~/glusterfs-${version}-?-*/build/glusterfs-${version}.tar.gz`)
+echo ${TGZS[0]}
+
+if [ -z ${TGZS[0]} ]; then
+        echo "wget https://download.gluster.org/pub/gluster/glusterfs/${series}/${version}/glusterfs-${version}.tar.gz"
+        wget https://download.gluster.org/pub/gluster/glusterfs/${series}/${version}/glusterfs-${version}.tar.gz
+else
+        echo "found ${TGZS[0]}, using it..."
+        cp ${TGZS[0]} .
+fi
+
+echo "Creating link file.."
+ln -s glusterfs-${version}.tar.gz glusterfs_${version}.orig.tar.gz
+
+echo "Untaring.."
+tar xpf glusterfs-${version}.tar.gz
+
+# Changelogs needed for building are maintained in a separate repo.
+# the repo has to be clone and updated properly so we can copy the changelogs so far.
+
+echo "Cloning the glusterfs-debian repo"
+git clone http://github.com/gluster/glusterfs-debian.git
+
+cd glusterfs-debian/
+
+git checkout -b ${flavor}-${series}-local origin/${flavor}-glusterfs-${series}
+
+if [ "$os" == "ubuntu" ]; then
+	sed -i "1i glusterfs (${version}-${os}1~${flavor}1) ${flavor}; urgency=medium\n\n * GlusterFS ${version} GA\n\n -- GlusterFS GlusterFS deb packages <deb.packages@gluster.org>  `date +"%a, %d %b %Y %T %z"`\n" debian/changelog
+elif [ "$os" == "debian" ]; then
+	sed -i "1i glusterfs (${version}-1) ${flavor}; urgency=low\n\n * GlusterFS ${version} GA\n\n -- Gluster Packager <glusterpackager@download.gluster.org>  `date +"%a, %d %b %Y %T %z"`\n" debian/changelog
+fi
+
+git commit -a -m "Glusterfs ${version} G.A (${flavor})"
+
+echo "Copying Changelog to source"
+cp -a debian ../glusterfs-${version}/
+
+echo "Building source package.."
+cd ../glusterfs-${version}
+debuild -S -sa -k${debuild_key}
+
+echo "Uploading the packages.."
+if [ "$os" == "ubuntu" ]; then
+        cd ..
+        dput ppa:gluster/glusterfs-${series} glusterfs_${version}-${os}1~${flavor}1_source.changes
+
+    echo "Done"
+    exit
```

Comment:
so, why are we exiting here, since the package is not build yet ?

Reply:
For Ubuntu, we are uploading the glusterfs_${version}-${os}1~${flavor}1_source.changes to [launchpad](https://launchpad.net/~gluster/+archive/ubuntu/glusterfs-7) Then the build process takes place on different architectures in the launchpad, which is automated by launchpad.

Reply:
So, this exist is where, our part in building process can be marked done for ubuntu.